tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala

```diff
@@ -129,9 +146,18 @@ class Z3SolverContext(val config: SolverConfig) extends SolverContext {
     val cellSort = getOrMkCellSort(cell.cellType)
     log(s"(declare-const $cell $cellSort)")
     val cellName = cell.toString
-    z3context.mkConstDecl(cellName, cellSort)
```

Comment:
why is it removed? Was it redundant?

Reply:
I believe so, `mkConstDecl` returns a declaration (see line 535 in the [Z3 API](https://github.com/Z3Prover/z3/blob/master/src/api/java/Context.java)) that is not asserted, so this line does not do anything. If you look at the original code of `declareInPredIfNeeded`, it also uses `mkConst`, but without `mkConstDecl`.

Reply:
Most likely, it is a left-over from the outdated code then